api/catalog/api/serializers/media_serializers.py

```diff
@@ -161,10 +175,30 @@ def validate_title(self, value):
     def validate_page_size(self, value):
         request = self.context.get("request")
         is_anonymous = bool(request and request.user and request.user.is_anonymous)
-        if is_anonymous and value > 20:
-            raise get_api_exception(
-                "Page size must be between 1 & 20 for unauthenticated requests.", 401
-            )
+        max_value = (
+            settings.MAX_ANONYMOUS_PAGE_SIZE
+            if is_anonymous
+            else settings.MAX_AUTHED_PAGE_SIZE
+        )
+
+        validator = MaxValueValidator(
+            max_value,
+            message=serializers.IntegerField.default_error_messages["max_value"].format(
+                max_value=max_value
+            ),
+        )
```

Comment:
Can be extracted into a separate validator, and applied to the field using the [`validators` argument](https://www.django-rest-framework.org/api-guide/fields/#validators). That removes the need for the `try`-`except` handling below.

Reply:
Great! We can do that on another refactor to provide a more specific error. Currently, it says `"Ensure this value is less than or equal to 20."` when before it was hinted that that was just the limit for unauthed requests: `"Page size must be between 1 & 20 for unauthenticated requests."`